acceptance/analyzer_test.go

```diff
@@ -0,0 +1,503 @@
+			when("DOCKER_CONFIG is set", func() {
+				it("succeeds", func() {
+					cmd := exec.Command(
+						"docker", "run", "--rm",
+						"--env", fmt.Sprintf("DOCKER_CONFIG=%s", registry.DockerDirectory),
+						"--network", "host",
+						analyzeImage,
+						analyzerPath,
+						"some-image",
+					)
+					h.Run(t, cmd)
+				})
+			})
+		})
+
+		when("cache is provided", func() {
+			when("cache image case", func() {
+				var cacheImage, authConfig string
+
+				it.Before(func() {
+					metadata := flattenMetadata(t, filepath.Join("testdata", "analyzer", "cache_metadata.json"), lifecycle.CacheMetadata{})
+					cacheImage, authConfig = buildRegistryImage(
+						t,
+						cacheImageRepo,
+						filepath.Join("testdata", "analyzer", "cache-image"),
+						"--build-arg", fmt.Sprintf("metadata=%s", metadata),
+					)
+				})
+
+				it.After(func() {
+					removeTestImage(t, cacheImage)
+				})
+
+				it("uses the provided cache", func() {
+					cmd := exec.Command(
+						"docker", "run", "--rm",
+						"--network", "host",
+						"--env", fmt.Sprintf("CNB_REGISTRY_AUTH=%s", authConfig),
+						analyzeImage,
+						"/bin/bash", "-c",
+						fmt.Sprintf("%s -cache-image %s some-image; ls -alR /layers", analyzerPath, cacheImage),
+					)
+					output := h.Run(t, cmd)
+
+					h.AssertMatch(t, output, "/layers/some-buildpack-id")
+					h.AssertMatch(t, output, "some-layer.sha")
+					h.AssertMatch(t, output, "some-layer.toml")
+				})
+			})
+
+			when("cache directory case", func() {
+				var (
+					cacheDir string
+					err      error
+				)
+
+				it.Before(func() {
+					cacheDir, err = ioutil.TempDir("", "analyzer-acceptance")
+					h.RecursiveCopy(t, filepath.Join("testdata", "analyzer", "cache-dir"), cacheDir)
+					// Some OSes (like macOS) use symlinks for the standard temp dir.
+					// Resolve it so it can be properly mounted by the Docker daemon.
+					cacheDir, err = filepath.EvalSymlinks(cacheDir)
+					h.AssertNil(t, err)
+				})
+
+				it.After(func() {
+					os.RemoveAll(cacheDir)
+				})
+
+				it("uses the provided cache", func() {
+					cmd := exec.Command(
+						"docker", "run", "--rm",
+						"--env", "CNB_REGISTRY_AUTH={}",
+						"--volume", fmt.Sprintf("%s:/cache", cacheDir),
+						analyzeImage,
+						"/bin/bash", "-c",
+						fmt.Sprintf("%s -cache-dir /cache some-image; ls -alR /layers", analyzerPath),
+					)
+					output := h.Run(t, cmd)
+
+					h.AssertMatch(t, output, "/layers/some-buildpack-id")
+					h.AssertMatch(t, output, "some-layer.sha")
+					h.AssertMatch(t, output, "some-layer.toml")
+				})
+			})
+		})
+	})
+
+	when("layers path and analyzed path are provided", func() {
+		it("writes analyzed.toml at the provided path", func() {
+			cmd := exec.Command(
+				"docker", "run", "--rm",
+				"--env", "CNB_REGISTRY_AUTH={}",
+				analyzeImage,
+				"/bin/bash", "-c",
+				fmt.Sprintf("%s -layers /other-layers -analyzed /other-layers/other-analyzed.toml some-image; ls -al /other-layers", analyzerPath),
```

Comment:
I find it a little odd that `-layers` must be used together with `-analyzed` (else we might run into permissions issues trying to write `/layers/analyzed.toml`). Also we're still expecting to find the group.toml at `/layers/group.toml`. I see how this all works but it's a bit confusing...

Reply:
I think that we set the `WORKDIR` in the Dockerfile to be `/layers` and we should give it `cnb` permissions.
I'm doing it in the detector acceptance tests with the following lines:
```
ARG cnb_uid=1234
ARG cnb_gid=1000

RUN groupadd cnb --gid ${cnb_gid} && \
  useradd --uid ${cnb_uid} --gid ${cnb_gid} -m -s /bin/bash cnb

RUN chown -R "${cnb_uid}:${cnb_gid}" "/layers"

WORKDIR /layers
```

Does it make sense?